### ISSUE TITLE
Save a session, with open buffer, command history and more

### ISSUE THREAD

**Issue by GuillaumeSeren** [pazz/alot#844]
Hi,
I heavily use alot and sometimes, due to a mistake or crash it close,
then I lost all my current buffer, eventually, current (to send) mail, and more (command history).

I think it could be really useful, to have a session, to save all that jazz, and also provide a more robust user experience.

What do you think, could it be useful to you ? Did I forget something ?

**Comment by pazz** [comment]
I agree that the ability to write info about
the current buffers and command history into a file, and commands to restore
such status, could be useful for some.
Due to the way I personally use alot (i close the UI once I'm done with it),
It's not a pressing priority for myself though.
Anyway, I welcome a discussion / suggestions about how to implement this nicely.

What file format would be easiest?
* ini-files
* line-by-line alot commands to restore?
* python pickling?



Quoting Guillaume Seren (2016-02-23 16:20:15)

**Comment by GuillaumeSeren** [comment]
As a vim user, I really like the vim's style of session handling, simple text file with vimscript command to return the state the way it was.
In the case of alot there is no vimscript but we, we can save queries, history, and other simple *cold* param, in a file.
So to me I prefer → "line-by-line alot commands to restore"

**Comment by lucc** [comment]
The history for different command prompts could be saved verbatim as lines into different files. For example one file for command prompt (`:`), one for all email address stuff (`t`, `c`, `f`, `b` in the envelope and the "to" prompt after `m`) and one for subject lines (after `m` or with `s` in envelope).
(This is much like mutt sorts the history and a feature that I have been missing in alot)

Restoring buffers and such was not a concern for me yet so no idea what would be best.

**Comment by GuillaumeSeren** [comment]
@lucc 
Yes I aggree on the simple verbatim side but not so sure ofr saving session into multiple files ( 1 by buffer right ? ), I'd like to avoid creating too much file automatically, specially on a ssd, i will check the session handling in mutt may be interesting if you have a more precise description of you usage I'll take it :wink:

**Comment by lucc** [comment]
The command line history and recipient history is in master now: #885 

related: #562

**Comment by GuillaumeSeren** [comment]
@lucc Thank you for the update.

I need a bit more time to focus on that, but what I would like to have 
* Save open buffers (at least the queries).
* -The command line history- Already done in #885

So it is 'only' about saving open buffers, I still have to figure this out,
but if we have already some session code merged in I need to check it first.
